test/instrumentation/decode_metric.go

```diff
@@ -104,7 +104,7 @@ func getMetricType(functionName string) string {
 }
 
 func (c *metricDecoder) decodeMetric(call *ast.CallExpr) (metric, error) {
-	if len(call.Args) != 1 {
+	if len(call.Args) > 2 {
```

Comment:
Looks like the restriction is less. Does it mean the length of call.Args is equal to 0 or 2 is legit or there is no such case? I am asking because the following code is returning the call.Args[0]...

Reply:
It turns out NewGaugeFunc takes two arguments but the opts struct is passed in the first argument, so the code just works if I do it like this.